### PR TITLE
Don't apply the spell check on all the file if no changes

### PR DESCRIPTION
**Pull request by sbrunner**
Before that:
```bash
$ time git commit --amend 2> log
real    7m19.811s
user    49m14.012s
sys     0m16.668s

$ ls -lh log
-rw-rw-r-- 1 sbrunner sbrunner 1,8M janv. 20 14:05 log
```

### CODE REVIEW

File: scripts/spell_check/check_spelling.sh [hunk 1]
```diff
@@ -41,6 +41,9 @@ shift $(expr $OPTIND - 1)
 if [ ! $# -eq 0 ]; then
   EXCLUDE=$(cat $AGIGNORE | sed -e 's/\s*#.*$//' -e '/^\s*$/d' | tr '\n' '|' | sed -e 's/|$//')
   INPUTFILES=$(echo $@ | tr -s '[[:blank:]]' '\n' | egrep -iv "$EXCLUDE" | tr '\n' ' ' )
+  if [ "$INPUTFILES" == "" ]; then
+    exit 0
+  fi
```
Comment (3nids) [inline scripts/spell_check/check_spelling.sh]: does it still work on all files when calling without argument?
I don't understand the aim of this check.

Comment (sbrunner) [inline scripts/spell_check/check_spelling.sh]: Yes, it you don't specify anything it's the test at line 41 who will going to select all the files, this test is useful when we specify only filtered files :-)

Comment (3nids) [inline scripts/spell_check/check_spelling.sh]: ok, thanks

**Comment by sbrunner** [comment]
Thanks :-)

**Pull request merged**